src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -607,12 +613,147 @@ CHIP_ERROR DefaultDACVerifier::VerifyNodeOperationalCSRInformation(const ByteSpa
+    VerifyOrReturnError(outSerialNumberHexStr.size() > serialNumber.size() * 2, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    Encoding::HexFlags flags = Encoding::HexFlags::kUppercaseAndNullTerminate;
+    err = BytesToHex(serialNumber.data(), serialNumber.size(), outSerialNumberHexStr.data(), outSerialNumberHexStr.size(), flags);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+
+    outSerialNumberHexStr.reduce_size(strlen(outSerialNumberHexStr.data()));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultDACVerifier::GetIssuerNameBase64Str(const ByteSpan & certDer, MutableCharSpan & outIssuerNameBase64String)
+{
+    uint8_t issuerBuf[kMaxCertificateDistinguishedNameLength] = { 0 };
+    MutableByteSpan issuer(issuerBuf);
+
+    CHIP_ERROR err = ExtractIssuerFromX509Cert(certDer, issuer);
+    VerifyOrReturnError(CHIP_NO_ERROR == err, err);
+    VerifyOrReturnError(outIssuerNameBase64String.size() > BASE64_ENCODED_LEN(issuer.size()), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    uint32_t encodedLen = Base64Encode32(issuer.data(), static_cast<uint32_t>(issuer.size()), outIssuerNameBase64String.data());
+    outIssuerNameBase64String.reduce_size(encodedLen);
+    return CHIP_NO_ERROR;
+}
+
+bool DefaultDACVerifier::IsCertificateRevoked(const ByteSpan & certDer)
+{
+    static constexpr uint32_t maxIssuerBase64Len = BASE64_ENCODED_LEN(kMaxCertificateDistinguishedNameLength) + 1;
+
+    char issuerNameBuffer[maxIssuerBase64Len]                                = { 0 };
+    char serialNumberHexStrBuffer[2 * kMaxCertificateSerialNumberLength + 1] = { 0 };
+    char akidHexStrBuffer[2 * kAuthorityKeyIdentifierLength + 1]             = { 0 };
+
+    MutableCharSpan issuerName(issuerNameBuffer);
+    MutableCharSpan serialNumber(serialNumberHexStrBuffer);
+    MutableCharSpan akid(akidHexStrBuffer);
+
+    CHIP_ERROR err = GetIssuerNameBase64Str(certDer, issuerName);
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, false);
+
+    err = GetSerialNumberHexStr(certDer, serialNumber);
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, false);
+
+    err = GetAKIDHexStr(certDer, akid);
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, false);
+
+    return IsEntryExistsInRevocationSet(akid, issuerName, serialNumber);
+}
+
 void DefaultDACVerifier::CheckForRevokedDACChain(const AttestationInfo & info,
                                                  Callback::Callback<OnAttestationInformationVerification> * onCompletion)
 {
     AttestationVerificationResult attestationError = AttestationVerificationResult::kSuccess;
 
-    // TODO(#33124): Implement default version of CheckForRevokedDACChain
+    if (mDeviceAttestationRevocationSetPath != nullptr)
+    {
+        if (IsCertificateRevoked(info.dacDerBuffer))
+        {
+            attestationError = AttestationVerificationResult::kDacRevoked;
+        }
+        if (IsCertificateRevoked(info.paiDerBuffer))
+        {
+            attestationError = AttestationVerificationResult::kPaiRevoked;
```

Comment:
So if both are revoked we consider the PAI being revoked more important than the DAC being revoked, in terms of the errors that get reported?

Reply:
Added another error code `kPaiAndDacRevoked` and using that one.